### DISCUSSION TITLE
Is it possible to dynamically mount directories?

### DISCUSSION THREAD

**Discussion by axeII** [ContainerSSH/ContainerSSH#299] [Questions] [answered]
Is it possible to dynamically mount directories?

Let's say the container-ssh host has a directory containing multiple subdirectories. I would like to set up that every container is dynamically mounted to a different sub-directory on the host based on e.g username.

Thanks for the reply.

**Comment by ghost** [comment, marked as answer]
Hello @axeII ,

Yes, the configuration server concept lets you do that. You will need to write a configuration server and configure it in ContainerSSH as described [here](https://containerssh.io/reference/configserver/). This configuration server then needs to return a config structure that tells the backend to mount the specified directory.

With the Docker backend for example, the webhook response would look like this:

```json
{
  "config": {
    "docker": {
      "execution": {
        "host": {
          "binds": [
            "/home/foo:/home/foo"
          ]
        }
      }
    }
  }
}
```

You can find more information about the possible Docker options [here](https://containerssh.io/reference/docker/) and about the Kubernetes options [here](https://containerssh.io/reference/kubernetes/).

Please let us know if you have more questions.
